rcljava/src/main/java/org/ros2/rcljava/action/ActionServer.java

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2020 ros2-java contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ros2.rcljava.action;
+
+import java.util.Collection;
+
+import org.ros2.rcljava.interfaces.Disposable;
+import org.ros2.rcljava.interfaces.ActionDefinition;
+
+public interface ActionServer<T extends ActionDefinition> extends Disposable {
```

Comment:
Yeah, it exactly like the "waitable" api from rclcpp and rclpy. Rather than introducing the waitable concept now, I figured it would be easier to introduce as a separate refactor. I've added a TODO in 6fe23e6

> it doesn't look like an ActionServer.

Yeah, not much is here now, but I presume we might add some stuff by the time we finish the implementation :shrug: